src/metaxy/metadata_store/base.py

```diff
@@ -583,11 +678,34 @@ def _validate_schema(self, df: pl.DataFrame) -> None:
                 f"DataFrame must have '{SNAPSHOT_VERSION_COL}' column"
             )
 
-    def _validate_schema_system_table(self, df: pl.DataFrame) -> None:
+    def _validate_schema_system_table(self, df: pl.DataFrame | pl.LazyFrame) -> None:
         """Validate schema for system tables (minimal validation)."""
         # System tables don't need metaxy_provenance_by_field column
         pass
 
+    def _apply_read_filters(
```

Comment:
I'm not sure if this is needed anymore, upstream dependency filtering has been moved to `metaxy.provenance` class